model/jpa/src/main/java/org/keycloak/connections/jpa/PersistenceExceptionConverter.java

```diff
@@ -27,31 +29,53 @@
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
 import java.lang.reflect.Proxy;
+import java.util.Arrays;
+import java.util.Collection;
 
 /**
  * @author <a href="mailto:sthorger@redhat.com">Stian Thorgersen</a>
  */
 public class PersistenceExceptionConverter implements InvocationHandler {
 
-    private EntityManager em;
+    private static final Collection<String> WRITE_METHOD_NAMES = Arrays.asList("persist", "merge");
 
-    public static EntityManager create(EntityManager em) {
-        return (EntityManager) Proxy.newProxyInstance(EntityManager.class.getClassLoader(), new Class[]{EntityManager.class}, new PersistenceExceptionConverter(em));
+    private final EntityManager em;
+    private final Boolean batchEnabled;
+    private final Integer batchSize;
```

Comment:
prefer primitive types when the values cannot be `null` so that the JVM can optimize more easily